src/Components/EnvironmentPicker/EnvironmentPicker.tsx

```diff
@@ -133,150 +141,131 @@ const EnvironmentPicker = ({
-                                )
-                        )?.id,
-                        storageAccountUrl: pair.storageAccountUrl,
-                        containerNames: [pair.containerName]
-                    };
-
-                    // loop the rest of the list to collect container names for that storage account url
-                    for (
-                        let i = idx + 1;
-                        i < storageAccountAndContainerList.length;
-                        i++
-                    ) {
-                        if (
-                            storageAccountAndContainerList[i]
-                                .storageAccountUrl === pair.storageAccountUrl
-                        ) {
-                            newPair.containerNames.push(
-                                storageAccountAndContainerList[i].containerName
-                            );
-                        }
-                    }
-                    storageAccountToContainersMapping.push(newPair);
-                }
-            });
-            defaultStorageAccountToContainersMappingsRef.current = storageAccountToContainersMapping;
-        } else if (storage?.containerUrl) {
-            const storageAccountAndContainer = getStorageAndContainerFromContainerUrl(
-                storage?.containerUrl
-            );
-            defaultStorageAccountToContainersMappingsRef.current = [
-                {
-                    storageAccountId: undefined,
-                    storageAccountUrl:
-                        storageAccountAndContainer.storageAccountUrl,
-                    containerNames: [storageAccountAndContainer.containerName]
-                } as StorageAccountToContainersMapping
-            ];
         }
-        adapter.setBlobContainerPath(storage.containerUrl);
-
-        const selectedStorageAccountUrl = getStorageAccountUrlFromContainerUrl(
-            storage.containerUrl
-        );
-        const storageAccountUrls =
-            defaultStorageAccountToContainersMappingsRef.current?.map(
-                (pair) => pair.storageAccountUrl
-            ) || [];
         environmentPickerDispatch({
             type: EnvironmentPickerActionType.SET_STORAGE_ACCOUNT_ITEMS,
             payload: {
                 storageAccountItems: {
-                    storageAccounts: storageAccountUrls,
-                    storageAccountToEdit: selectedStorageAccountUrl
+                    storageAccounts,
+                    storageAccountToEdit
                 }
             }
         });
+        // END of initializing Storage account url options
+
+        // START of initializing Storage container name options
+        let containers: Array<string> = [];
+        let containerToEdit: string;
+        if (storage?.containerUrl) {
+            containerToEdit = getContainerNameFromUrl(storage?.containerUrl); // passed prop has precedence over the localstorage, if enabled
```

Comment:
Same as above for this if / if combination

Reply:
these logics are couple of lines down below because they are tied to storage.isLocalStorageEnabled prop. Please check L#226